modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/event/ApplicationClusterEventListener.java

```diff
@@ -17,100 +21,85 @@
 public class ApplicationClusterEventListener
     implements EventListener
 {
-    private ApplicationService applicationService;
+    private static final Logger LOG = LoggerFactory.getLogger( ApplicationClusterEventListener.class );
+
+    private final ApplicationService applicationService;
+
+    @Activate
+    public ApplicationClusterEventListener( @Reference final ApplicationService applicationService )
+    {
+        this.applicationService = applicationService;
+    }
 
     @Override
     public void onEvent( final Event event )
     {
-        if ( event != null )
+        if ( !event.isLocalOrigin() )
         {
-            if ( event.isLocalOrigin() )
-            {
-                return;
-            }
-
             doHandleEvent( event );
         }
     }
 
     private void doHandleEvent( final Event event )
     {
-        final String type = event.getType();
-
-        if ( type.equals( ApplicationClusterEvents.EVENT_TYPE ) )
+        if ( ApplicationClusterEvents.EVENT_TYPE.equals( event.getType() ) )
         {
-            final String eventSubType = event.getValueAs( String.class, ApplicationClusterEvents.EVENT_TYPE_KEY ).get();
-
-            switch ( eventSubType )
-            {
-                case ApplicationClusterEvents.INSTALLED:
-                    handleInstalledEvent( event );
-                    break;
-                case ApplicationClusterEvents.STATE_CHANGE:
-                    handleStateChangedEvent( event );
-                    break;
-                case ApplicationClusterEvents.UNINSTALLED:
-                    handleUninstalledEvent( event );
-                    break;
-            }
+            event.getValueAs( String.class, ApplicationClusterEvents.EVENT_TYPE_KEY ).
+                ifPresent( eventSubType -> {
+                    switch ( eventSubType )
```

Comment:
Codacy found an issue: [A switch statement does not contain a break](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=6220523)